src/Bicep.Core/TypeSystem/ResourceType.cs

```diff
@@ -11,20 +12,24 @@ public record ResourceTypeComponents(
 
     public class ResourceType : TypeSymbol, IScopeReference
     {
-        public ResourceType(NamespaceType declaringNamespace, ResourceTypeReference typeReference, ResourceScope validParentScopes, ITypeReference body)
+        public ResourceType(NamespaceType declaringNamespace, ResourceTypeReference typeReference, ResourceScope validParentScopes, ITypeReference body, ImmutableHashSet<string> uniqueIdentifierProperties)
```

Comment:
It looks like these are a list of properties at the top level of the resource. How will this work for Kubernetes? `.metadata.name` and `.metadata.namespace`?

Reply:
It's not 100% necessary for the kubernetes prototype - it's only useful for the 'duplicate resource' check, which I'm currently skipping for non-ARM resource types.